### ISSUE TITLE
Ubuntu 22.04 LTS

### ISSUE THREAD

**Issue by sonyclar** [libretime/libretime#1845]
Ubuntu 22.04 LTS is not supported with "ubuntu-22.04"

![image](https://user-images.githubusercontent.com/64663853/167244314-165f07e7-1086-48e8-b9b9-c211aa35c407.jpeg)

Did i wrong something?

**Comment by paddatrapper** [comment]
Support hasn't been added yet for 22.04

**Comment by jooola** [comment]
Ubuntu ships the following dependencies that LibreTime does not support:
- Php 8.1 (LibreTime expect Php >=7.2, <8.0)
- Liquisoap 2.0 (LibreTime expect Liquisoap >= 1.1.1, < 2.0 0)
- there might be other dependencies that aren't supported that we don't know of.

If you find a way to go around thoses dependencies you might has a small chance to run LibreTime on Ubuntu 22.04. Do this at your own risk.

**Comment by kaalvoetjaco** [comment]
Thanks for all the hard work with this project. Are you perhaps planning on adding support for Ubuntu 22.04 in the near future?

**Comment by paddatrapper** [comment]
We intend to add support, but there is no timeline yet

**Comment by maxtimbo** [comment]
In the meantime, it would be helpful to mention that on LibreTime.org as it mentions "Ubuntu Current LTS". Which it was my assumption that 22.04 would be that version.

**Comment by sonyclar** [comment]
@maxtimbo exactly. I've tried to install LibreTime on Ubuntu 22.04 because it's LTS version.

**Comment by jooola** [comment]
A branch we should keep an eye on, for Propel: https://github.com/tworzenieweb/Propel/tree/feature/php8.0

**Comment by chlowden** [comment]
Hello, Just wondering if the dev team had any news on validating 22.4 by any chance? Many thanks

**Comment by paddatrapper** [comment]
Not yet. There is no eta at this point unfortunately

**Comment by chlowden** [comment]
Thank you for all your hard work.

**Comment by clintguy74** [comment]
will there be any chance this willhappen in the near future, ubuntu 24 is already stable, and in my country the repository for ubuntu 20.04 is constantly dropping, I was trying to move from Azuracast, but it seems I may have to stay, 
Libre time looks great, but cant get the repository to work, doesnt help, no mater what, even changing DNS.  Im in south africa by the way.

**Comment by epj** [comment]
Time is running out on 20.04 standard support.  I need a path to a newer Ubuntu.  Can I update Ubuntu in place to 22.04 and Libretime still works?

**Comment by explorier** [comment]
Is there any progress update on getting to a recent version of Ubuntu LTS? Is there a way external developers can help with this? Being stuck on Ubuntu 20 is a major issue since support ends after May this year.

**Comment by paddatrapper** [comment]
Support for liquidsoap v2 has been added. The issue is still PHP 7.4. We could potentially install it from https://launchpad.net/~ondrej/+archive/ubuntu/php and then use that. Upgrading to PHP 8 is a large process that requires someone with understanding of the PHP code base to do

**Comment by chlowden** [comment]
I am no php specialist ... but I really would like to migrate my libretime instance to a 22.4 instance. Is the issue that the community does not know anyone able to do what is necessary or does not have the means to get this know how to work on it? Would launching a crowdfunding endeavour help to move this forward? I would happily contribute if this is what is needed to keep this marvellous project relevant.

**Comment by explorier** [comment]
> Support for liquidsoap v2 has been added. The issue is still PHP 7.4. We could potentially install it from https://launchpad.net/~ondrej/+archive/ubuntu/php and then use that. Upgrading to PHP 8 is a large process that requires someone with understanding of the PHP code base to do

Are there other dependencies that would pose issues or be a roadblock to 22 support that you're aware of?

Utilizing and installing PHP 7.4 on Ubuntu 22.04 should work just fine, right? PPA approach seems standard.

If there's not any other roadblocks to 22.04 usage, and if Libretime 4.3.0 theoretically runs on 22.04, I would be interested in possibly doing some work on the [Libretime Ansible role](https://github.com/libretime/ansible-role-libretime) to support that installation. I spent some hours last night attempting it, and although my custom wrapper around the Libretime role did successfully run and seemed to install, I abandoned ship after not successfully getting the Libretime services running - there were just too many open questions for me to justify more work on it without direct knowledge or documentation of what was preventing the project from supporting Jammy. So for now I'm just going to incorporate the existing role to install on an Ubuntu 20 server and rely on Ubuntu Pro extended support.

I'm also curious about the PHP 8 upgrade and supporting that, I work on other PHP application codebases and have done (much smaller) upgrades from 7 -> 8. It seems like a valuable project for Libretime to prioritize for many reasons, although I understand it can be quite difficult for a large codebase with many dependencies.

Anyway, thanks for your quick response, and all your work! This is a great project our org has utilized for years, and I'd love to learn more and see it thrive.

**Comment by paddatrapper** [comment]
> Are there other dependencies that would pose issues or be a roadblock to 22 support that you're aware of?

The dependencies that immediately spring to mind are:

1. PHP 7.4 - can probably use the PPA as a workaround (untested)\
2. Python 3.6 - I'm not sure there is anything required to update here, but it is untested.
3. Celery 4 - we probably need to update Celery to v5, which requires updating the PHP and Python libraries used.

> Utilizing and installing PHP 7.4 on Ubuntu 22.04 should work just fine, right? PPA approach seems standard.

Yes, but would need testing

> I'm also curious about the PHP 8 upgrade and supporting that, I work on other PHP application codebases and have done (much smaller) upgrades from 7 -> 8. It seems like a valuable project for Libretime to prioritize for many reasons, although I understand it can be quite difficult for a large codebase with many dependencies.

Our biggest blocker is the Propel v1 framework that forms the basis of pretty much all the PHP used in LibreTime. We have forked the repo (https://github.com/libretime/propel1) because it is dead upstream, but it definitely doesn't support PHP 8.

**Comment by chlowden** [comment]
> We have an [OpenCollective](https://opencollective.com/libretime) which people can donate to. Those funds were initially set up for covering the project's hosting fees (demo instance, domain, email, etc). There was no consensus on using them to fund any development. However, PHP 7.4 is EOL and updating is a priority. So I would support using those funds to update to PHP 8.x, as we are not going to have a new frontend in any fast time to replace it. The issue is finding someone willing and capable to do it.

I have chipped in a contribution that you can use as you see fit ... but I think that php 8 is really the ambition to have. I believe it can give a new spring of life in the Libretime project.

**Comment by paddatrapper** [comment]
@chlowden thanks! Yes definitely. I would have liked to replace the PHP completely instead of updating, but we've run out of time for that unfortunately

**Comment by SteampunkEnterprise** [comment]
Hi, Any update on support for Ubuntu 22.04 LTS yet with 20.04 end of life?
Has any further progress on PHP 8?

**Comment by chlowden** [comment]
> We have forked the repo (https://github.com/libretime/propel1) because it is dead upstream, but it definitely doesn't support PHP 8.

I am sure you know this ... but for what it is worth there is Propel2 
https://github.com/propelorm/Propel2/
They are using it with php 8
